### PR TITLE
Small note to describe Nil

### PR DESCRIPTION
**Pull request by manishpatelUK**
This is the first time the user comes across Nil in the tutorial, so a small explanation of what it is.

### CODE REVIEW

File: _tour/generic-classes.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ class Stack[A] {
 ```
 This implementation of a `Stack` class takes any type `A` as a parameter. This means the underlying list, `var elements: List[A] = Nil`, can only store elements of type `A`. The procedure `def push` only accepts objects of type `A` (note: `elements = x :: elements` reassigns `elements` to a new list created by prepending `x` to the current `elements`).
 
+`Nil` here is a special case of `List` and is not to be confused with `Null`. It is a `List` of zero length.
```
Comment (Philippus) [inline _tour/generic-classes.md]: Let's mention the term "the empty list".

Comment (manishpatelUK) [reply _tour/generic-classes.md]: Done

File: _tour/generic-classes.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ class Stack[A] {
 ```
 This implementation of a `Stack` class takes any type `A` as a parameter. This means the underlying list, `var elements: List[A] = Nil`, can only store elements of type `A`. The procedure `def push` only accepts objects of type `A` (note: `elements = x :: elements` reassigns `elements` to a new list created by prepending `x` to the current `elements`).
 
-`Nil` here is a special case of `List` and is not to be confused with `Null`. It is a `List` of zero length.
+`Nil` here is a special case of `List` and is not to be confused with `Null`. It is simply an empty list.
```
Comment (martijnhoekstra) [inline _tour/generic-classes.md]: Maybe just "`Nil` here is the empty `List` (not to be confused with `null`)." is enough here.

Comment (manishpatelUK) [reply _tour/generic-classes.md]: Same difference, IMO, but now changed

File: _tour/generic-classes.md [hunk 3]
```diff
@@ -31,6 +31,8 @@ class Stack[A] {
 ```
 This implementation of a `Stack` class takes any type `A` as a parameter. This means the underlying list, `var elements: List[A] = Nil`, can only store elements of type `A`. The procedure `def push` only accepts objects of type `A` (note: `elements = x :: elements` reassigns `elements` to a new list created by prepending `x` to the current `elements`).
 
+`Nil` here is an empty `List` and is not to be confused with `Null`.
```
Comment (SethTisue) [inline _tour/generic-classes.md]: `Nil` is a term, not a type, so I think the term `null` would make more sense here, `Null` is the type.

Comment (Philippus) [inline _tour/generic-classes.md]:
```suggestion
`Nil` here is an empty `List` and is not to be confused with `null`.
```

**Comment by SethTisue** [comment]
final fixup in bae05563ec3b36db888b63edee0d3c165e9f3d65

**Pull request merged**